subworkflows/local/utils_nfcore_mcmicro_pipeline/main.nf

```diff
@@ -148,6 +150,121 @@ workflow PIPELINE_COMPLETION {
     }
 }
 
+workflow UPDATE_FROM_OME {
+    take:
+    samplesheet
+    markersheet
+
+    main:
+    samplesheet.map{meta, image_tiles, dfp, ffp -> [meta, image_tiles]} | OMEXTRACTOR
+
+    val_data = OMEXTRACTOR.out.xml | OMEVALIDATION
+
+    samplesheet.join(val_data)
+         .map { original_meta, image_tiles, dfp, ffp, xml_meta, marker_data ->
+                [xml_meta + original_meta, image_tiles, dfp, ffp]
+        }
+        .dump(tag: "ch_samplesheet_meta")
+        .set { samplesheet_meta }
+
+    c_sum = 0
+    agg = channel.empty()
+
+    samplesheet_meta
+        .map {
+            meta, image_tiles, dfp, ffp -> [meta.cycle_number, meta.channel_count]
+        }
+        .unique()
+        .toSortedList()
+        .flatMap()
+        .map {
+            cn, cc ->
+            temp = [cn, c_sum]
+            c_sum += cc
+            return temp
+        }
+        .dump(tag: "CHANNEL_DELTA_INDEX")
+        .set { agg }
+
+    // update val_data channel_number so it matches with samplesheet
+    val_data.map {
+            meta, xml_meta, marker_meta -> marker_meta
+        }
+        .flatten()
+        .map{
+            meta -> [meta.cycle_number, meta]
+        }
+        .combine(agg, by: 0)
+        .map {
+            key, meta, counter ->
+            meta.channel_number += counter
+            return meta
+        }
+        //.dump(tag:"vd")
+        .set{ val_data }
+
+
+    markersheet_template = //markersheet.flatten().first().keySet().collectEntries {key -> [key, null]}.toList().first().dump(tag: "template")
+    [
+        'channel_number':null, 'cycle_number':null,
+        'marker_name':null, 'filter':null,
+        'excitation_wavelength':null, 'emission_wavelength':null,
+        'exposure':null, 'background':null,
+        'remove':null
+        ]
+
+    markersheet
+        .flatten()
+        .dump(tag: "ch_markersheet_premeta")
+
+        .map {
+            entry ->
+            [['channel_number': entry.channel_number, 'cycle_number': entry.cycle_number], entry.findAll {it.value != null}]
```

Comment:
The map expression might be written more concisely as `entry.subMap('channel_number', 'cycle_number')'. Same goes for this pattern repeated below.